daemon/rpc_connect.go

```diff
@@ -44,6 +45,34 @@ func (r *RPC) Connect(in *pb.ConnectRequest, srv pb.Daemon_ConnectServer) (retEr
 	return err
 }
 
+// quenchConfigFallback checks if technology is configured to quench and falls back to NordLynx if quench was disabled
+// in compile time or in remote config. It returns a new config with desired technology set.
+func (r *RPC) quenchConfigFallback(cfg config.Config) config.Config {
+	if cfg.Technology != config.Technology_QUENCH {
+		return cfg
+	}
+
+	quenchEnabled, err := r.remoteConfigGetter.GetQuenchEnabled(r.version)
+	if err != nil {
+		log.Println(internal.ErrorPrefix, "failed to retrieve remote config for quench:", err)
+	}
+
+	if features.QuenchEnabled && (quenchEnabled && err == nil) {
+		return cfg
+	}
+
+	log.Println(internal.DebugPrefix,
+		"user had configured Quench technology, but it was disabled, falling back to NordLynx")
+
+	cfg.Technology = config.Technology_QUENCH
```

Comment:
Should it be `NORDLYNX` here? otherwise where is promised fallback to NORDLYNX?

Reply:
also if the technology is changed for the user it might be good to print a message about this? Otherwise it might be confusing.